pkg/controller/plan/adapter/ovirt/builder.go

```diff
@@ -394,9 +394,27 @@ func (r *Builder) mapCPU(vm *model.Workload, object *cnv.VirtualMachineSpec) {
 	}
 	if vm.CustomCpuModel != "" {
 		r.setCpuFlags(vm.CustomCpuModel, object)
+	} else if r.Plan.Spec.PreserveClusterCPUModel {
+		r.setCpuFlags(r.getClusterCpu(vm), object)
 	}
 }
 
+func (r *Builder) getClusterCpu(vm *model.Workload) string {
+	clusterVersion := vm.Cluster.Version.Major + "." + vm.Cluster.Version.Minor
+	for _, serverCpuOption := range vm.ServerCpu.SystemOptionValue {
+		if serverCpuOption.Version != clusterVersion {
+			continue
+		}
+		cpus := strings.Split(serverCpuOption.Value, ";")
+		for _, values := range cpus {
+			if strings.Contains(values, vm.Cluster.CPU.Type) {
+				return strings.Split(values, ":")[3]
+			}
+		}
+	}
+	return ""
```

Comment:
this can't happen, right? if it can't happen, I'd declare a variable before the for-loop and update it within the for-loop. this way it seems like we expect to return an empty string in some cases..